homeassistant/components/freebox/number.py

```diff
@@ -0,0 +1,130 @@
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the switch."""
+    router: FreeboxRouter = hass.data[DOMAIN][entry.unique_id]
+    entities = [
+        FreeboxLCDNumber(router, entity_description)
+        for entity_description in LCD_NUMBER_DESCRIPTIONS
+    ]
+    async_add_entities(entities, True)
+
+
+class FreeboxNumber(NumberEntity):
+    """Representation of a Freebox number entity."""
+
+    entity_description: FreeboxNumberEntityDescription
+
+    def __init__(
+        self,
+        router: FreeboxRouter,
+        entity_description: FreeboxNumberEntityDescription,
+    ) -> None:
+        """Initialize the switch."""
+        self.entity_description = entity_description
+        self._attr_name = f"Freebox {entity_description.name}"
+        self._attr_unique_id = f"{router.mac}-{entity_description.key}"
+        self._attr_device_info = router.device_info
+        self._router = router
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Set the native value of the entity."""
+        await self.entity_description.async_set_value(self._router, int(value))
+
+
+class FreeboxLCDNumber(FreeboxNumber):
+    """Representation of a Freebox number entity."""
+
+    @callback
+    def async_update_state(self) -> None:
+        """Update the Freebox sensor."""
+        self._attr_native_value = self._router.lcd_settings.get(
+            self.entity_description.key
+        )
+
+    @callback
+    def async_on_demand_update(self):
+        """Update state."""
+        self.async_update_state()
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Register state update callback."""
+        self.async_update_state()
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                self._router.signal_lcd_update,
+                self.async_on_demand_update,
+            )
+        )
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Set the native value of the entity."""
+        try:
+            await super().async_set_native_value(value)
+
+            self._router.lcd_settings[self.entity_description.key] = int(value)
+
+            self.async_update_state()
+            self.async_write_ha_state()
+        except InsufficientPermissionsError:
+            self.async_on_demand_update()
+
+            # Send notification
+            self.hass.components.persistent_notification.async_create(
+                'Home Assistant does not have permission to update the LCD settings.\nYou should grant the "Edit the Freebox\'s settings" permission in the Freebox OS web interface.',
+                title="Missing permission - Freebox",
+                notification_id="freebox_missing_lcd_permission",
+            )
```

Comment:
This should not create a persistent notification. Those are generally for end-user. Please raise an issue in the repair dashboard instead.